src/components/Table/TableHeaderComponent.jsx

```diff
@@ -73,7 +102,9 @@ TableHeader.contextTypes = {
 
 TableHeader.propTypes = {
   visibleColumns: PropTypes.array.isRequired,
-  properties: PropTypes.object.isRequired
+  properties: PropTypes.object.isRequired,
+  checkedAll: PropTypes.object.isRequired,
+  checkAll: PropTypes.func.isRequired
```

Comment:
`checkedAll` is not a simple flag because it's used both by header and row. Without a flag `changedByRow` there is a problem with (un)checking a simple row.

I'm considering another names.